components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -50,6 +53,11 @@ func (s *OIDCService) CreateClientConfig(ctx context.Context, req *connect.Reque
 		return nil, err
 	}
 
+	err = assertIssuerIsReachable(req.Msg.GetConfig().GetOidcConfig().GetIssuer())
```

Comment:
Does it make sense to check the URL actually parses as a URL, before we make this request?

How do we make sure this doesn't get used as a way to DDOS third party systems? If I supply a `https://my-victim.org`, we'd hit those endpoints on behalf of the attacker.

One way to guard would be to rate-limit the create itself.

Reply:
+1 for rate-limiting.